[triton 3.3] Fix aoti cpp wrapper remaining 5 issue. (following #148051)

YUNQIUGUO opened the pull request:
Summary:
Fix the following 5 on a100:

- test_foreach_cpp_wrapper_cuda_gpu_wrapper

- test_enable_dynamic_shapes_cpp_wrapper_cuda_gpu_wrapper

- test_dynamic_shapes_persistent_reduction_mixed_x_dim_cuda_gpu_wrapper

- test_enable_dynamic_shapes_cpp_wrapper_cuda_dynamic_shapes_gpu_wrapper

- test_dynamic_shapes_persistent_reduction_mixed_x_dim_cuda_dynamic_shapes_gpu_wrapper

Test Plan:
oss :

```
TORCHINDUCTOR_COMPILE_THREADS=1 TORCHINDUCTOR_MAX_AUTOTUNE_GEMM_BACKENDS=TRITON TORCH_LOGS="+inductor, output_code" TORCHINDUCTOR_FORCE_DISABLE_CACHES=1 CPLUS_INCLUDE_PATH=/usr/local/cuda-12.6/include:$CPLUS_INCLUDE_PATH python test/inductor/test_gpu_cpp_wrapper.py -k test_foreach_cpp_wrapper_cuda_gpu_wrapper
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

YUNQIUGUO commented:
issue: https://github.com/pytorch/pytorch/issues/148111

davidberard98 commented on torch/_inductor/codegen/cpp_wrapper_gpu.py:
```diff
@@ -571,26 +571,31 @@ def generate_kernel_call(
-                    for arg_type, arg_name in zip(arg_types, signature)
-                    if signature[arg_name] != "constexpr"
-                ]
+                if (
+                    triton_meta is not None
+                    and triton_meta.get("configs")
+                    and triton_meta.get("signature")
+                ):
```
minor nit: could we put this check on the outside (if triton_meta is not None...) and put the `triton_version_uses_attrs_dict` check inside that?

YUNQIUGUO replied:
good nit!

davidberard98 reviewed (approved):
Nice work!! thanks for the quick fix

YUNQIUGUO committed:
[triton 3.3] Fix aoti cpp wrapper remaining 5 issue. (following #148051) (#148117)

Summary:

Fix the following 5 on a100:

- test_foreach_cpp_wrapper_cuda_gpu_wrapper

- test_enable_dynamic_shapes_cpp_wrapper_cuda_gpu_wrapper

- test_dynamic_shapes_persistent_reduction_mixed_x_dim_cuda_gpu_wrapper

- test_enable_dynamic_shapes_cpp_wrapper_cuda_dynamic_shapes_gpu_wrapper

- test_dynamic_shapes_persistent_reduction_mixed_x_dim_cuda_dynamic_shapes_gpu_wrapper

Test Plan:
oss :

```
TORCHINDUCTOR_COMPILE_THREADS=1 TORCHINDUCTOR_MAX_AUTOTUNE_GEMM_BACKENDS=TRITON TORCH_LOGS="+inductor, output_code" TORCHINDUCTOR_FORCE_DISABLE_CACHES=1 CPLUS_INCLUDE_PATH=/usr/local/cuda-12.6/include:$CPLUS_INCLUDE_PATH python test/inductor/test_gpu_cpp_wrapper.py -k test_foreach_cpp_wrapper_cuda_gpu_wrapper
```

Reviewed By: davidberard98

Differential Revision: D70346620

chenyang78 reviewed (approved):
Nice! Thanks for the quick fix.

davidberard98 commented:
@pytorchbot merge -f "already landed in fbcode, and macos jobs are really unlikely to be affected by this"

davidberard98 commented:
@wdvr do you know what's happening here?

wdvr commented:
@pytorchbot merge -f "already landed in fbcode"

wdvr commented:
@pytorchbot merge -f "already landed in fbcode"

(trying now with the DIFF removed from the first comment)
